File: prototype/java/starcraft/src/main/java/application/Application.java
package application;

import lines.model.Unit;
import lines.model.UnitAction;
import lines.service.domain.MarineAction;
import lines.service.factory.UnitFactory;

public class Application {
    public static void main(String[] args) {
        UnitAction unitAction = new MarineAction();

        for (int i = 0; i < 60; i++) {
            Unit unit = UnitFactory.CreateUnit(unitAction);

            unit.Attack();

            unit.Building();

            unit.Harvest();
        }
    }
}

File: prototype/kotlin/starcraft/src/main/kotlin/lines/application/Application.kt
package lines.application

import lines.service.domain.MarineAction
import lines.service.factory.UnitFactory

fun main(){
    val marineAction = MarineAction()

    for (i in 1..30) {
        val createUnit = UnitFactory.CreateUnit(marineAction)

        createUnit.Attack()

        createUnit.Building()

        createUnit.Harvest()
    }

}